rust/osm2lanes/src/transform/tags_to_lanes/separator/semantic.rs

```diff
@@ -0,0 +1,79 @@
+use crate::metric::{Metre, Speed};
+use crate::road::{Designated, Style};
+use crate::transform::tags_to_lanes::Infer;
+
+/// Semantic speed class
+pub enum SpeedClass {
+    Walking,
+    /// ~30kph / ~20mph
+    Living,
+    /// ~50kph / ~30mph
+    Intra,
+    /// ~80kph / ~50mph
+    Inter,
+    Max,
+}
+
+impl From<Speed> for SpeedClass {
+    #[allow(clippy::panic)]
+    fn from(s: Speed) -> Self {
+        match s.kph() {
+            s if (0.0_f64..15.0_f64).contains(&s) => Self::Walking,
+            s if (15.0_f64..40.0_f64).contains(&s) => Self::Living,
+            s if (40.0_f64..70.0_f64).contains(&s) => Self::Intra,
+            s if (70.0_f64..100.0_f64).contains(&s) => Self::Inter,
+            s if (100.0_f64..200.0_f64).contains(&s) => Self::Max,
+            _ => panic!("unexpected speed {s}"),
+        }
+    }
+}
+
+/// Overtaking rules
+///
+/// Note: this does not take into account the local vehicle-specific rules,
+/// e.g. for motorcycle filtering or overtaking tractors.
+pub enum Overtaking {
+    Permitted,
+    _Prohibited,
+}
+
+impl Default for Overtaking {
+    fn default() -> Self {
+        // fail-deadly, see README
+        Self::Permitted
+    }
+}
+
+/// Semantic lane separator
+pub enum Separator {
+    /// Into grass or dirt
+    _SoftEdge,
+    /// Into a building or other hard surface
+    _HardEdge,
+    /// Motorway (or other) shoulder
+    Shoulder { speed: Infer<SpeedClass> },
+    /// Road paint between same direction
+    Lane {
+        speed: Infer<SpeedClass>,
+        overtaking: Overtaking,
+    },
+    /// Road paint between opposite direction
+    Centre {
```

Comment:
Maybe the variants that are just paint should have "Marking" in their names? Like `CenterMarking`, `LaneMarking`, `MarkedBuffer`, and `ShoulderMarking` (or `EdgeMarking`), to avoid confusion with other types and remind us that physical separators will use a different variant.

Reply:
I could also nest the enum:

```
    ...
    Marking(Marking),
    ...

    enum Marking {
        Centre,
        Lane,
    }
```

Reply:
Hmm.. I think I prefer a flat enum. `Separator::Centre` is also clear, but my novice ruse instinct was to `use Separator::*` in my head :P I would not object to whichever you choose.